Sources/RealModule/ApproximateEquality.swift

```diff
@@ -128,13 +208,20 @@ extension Numeric where Magnitude: FloatingPoint {
   ///     ```
   ///     (if we used the default, we would be testing if `z` were inside a square region
   ///     instead.)
-  @inlinable @inline(__always)
-  public func isApproximatelyEqual(
+  @inlinable
+  public func isApproximatelyEqual<Magnitude>(
     to other: Self,
     absoluteTolerance: Magnitude,
     relativeTolerance: Magnitude = 0,
-    norm: (Self) -> Magnitude = { $0.magnitude }
-  ) -> Bool {
+    norm: (Self) -> Magnitude
+  ) -> Bool
+  // TODO: constraint should really be weaker than FloatingPoint,
+  // but we need to have `isFinite` for it to work correctly with
+  // floating-point magnitudes in generic contexts, which is the
+  // most common case. The fix for this is to lift the isFinite
+  // requirement to Numeric in the standard library, but that's
+  // source-breaking, so requires an ABI rumspringa.
```

Comment:
Maybe. That would work for floating point, but there's no semantic guarantee that it actually does something sensible for an unconstrained type.

Reply:
(There's also another issue in making this too general, which is that the computation of `delta` itself can trap for most integer-derived types; because of that I am loosely leaning towards viewing the Magnitude = FloatingPoint constraint as a good thing here.)

Reply:
I think it would be reasonable to document some degree of semantic restriction on the 'exceptional' values that `Numeric` can support. I don't know if there'd be any problem with explicitly limiting them to IEEE 754 NaN-or-infinity; can you think of any problems that would arise with that formulation?

Reply:
I don't see an big problem with it, but we'd have to define "IEEE 754 nan-or-infinity" in some way that makes sense for more general types (e.g. would `Complex`'s single non-finite value fit into that?)

Reply:
Using `delta - delta == 0` would also explicitly rule-out ever supporting running with the invalid exception unmasked, which is generally not a great plan but some scientific software written in C and Fortran wants to do it for complicated historical reasons (if Swift were to support IEEE 754 exceptions someday, infinity.isFinite should not raise invalid, but `infinity - infinity` should).